blurp-app/src/pages/blurpmap.jsx

```diff
@@ -49,32 +51,169 @@ const TestPage = () => {
   const [edgeData, setEdgeData] = useState({ familiarity: 0, stressCode: STRESS_CODE.MINIMAL });
   const [pos, setPos] = useState({ x: 0, y: 0 });
   const [sigma, setSigma] = useState(null);
-  const child = useRef();
   const [clickTrigger, setClickTrigger] = useState(true);
+  const child = useRef();
+  const instance = axios.create({
+    timeout: 1000,
+  });
+
   // Used for the message box that pops up and notifys users of errors
   const [userNotification, setUserNotification] = useState('');
   const msgRef = useRef();
 
+  // Temporary db userID/mapID for testing
+  const [profile, setProfile] = useState({
+    profileSet: false,
+    userID: 'bb9e434a-7bb9-493a-80b6-abafd0210de3',
+    mapID: '71328e4f-15b6-4189-99a5-dca424b1fea8',
+  });
+
   const DBref = useRef({
     SaveToDB() {
-      console.log(JSON.stringify(graph));
-      axios
-        .get('http://localhost:3000/')
-        .then((response) => {
-          console.log(response);
-        })
-        .catch((error) =>
-          console.log(
-            'ERROR: ' +
-              error.message +
-              '\n' +
-              'Failed to communicate with database\n' +
-              'error name: ' +
-              error.name +
-              'request name: ' +
-              error.request
-          )
-        );
+      if (profile.profileSet) {
+        graph.forEachNode((current, attr) => {
+          if (current) {
+            instance
+              .post(BACKEND_URL + '/map/node/create', {
+                userID: profile.userID,
+                mapID: profile.mapID,
+                nodeinfo: {

```

Comment:
can this possibly cause null?

Reply:
I came up with a solution where it will set the value to 0 upon saving to db, and will set years to '' upon loading it in and seeing 0. I don't really like this as this prevents from entering 0 as a value. Perhaps not have it allow negative values and instead set it to -1 to signal that the user hasn't filled it out yet?